modules/contract/contracts.js

```diff
@@ -149,9 +162,16 @@ class Contracts {
                                       fee, '', storageLimit, gasLimit,
                                       contract, initValue, conseiljs.TezosParameterFormat.Michelson);        
                                       
-            if(!tezosNode.includes('localhost') && !tezosNode.includes('192.168')) {
+            if(Helper.isCarthagenetNode(tezosNode)) {
+                try {
+                    await conseiljs.TezosConseilClient.awaitOperationConfirmation(carthagenetConseilServer, carthagenetConseilServer.network, JSON.parse(result.operationGroupID), 15, 10);
+                } catch(error) {
+                    Helper.errorLogHandler(`Error occurred in operation confirmation: ${error}`,
+                                           'Contract deployment operation confirmation failed ....');
+                }
+            } else if(Helper.isMainnetNode(tezosNode)) {
                 try {
-                    await conseiljs.TezosConseilClient.awaitOperationConfirmation(conseilServer, conseilServer.network, JSON.parse(result.operationGroupID), 15, 10);
+                    await conseiljs.TezosConseilClient.awaitOperationConfirmation(mainnetConseilServer, mainnetConseilServer.network, JSON.parse(result.operationGroupID), 15, 10);
                 } catch(error) {
                     Helper.errorLogHandler(`Error occurred in operation confirmation: ${error}`,
                                            'Contract deployment operation confirmation failed ....');
```

Comment:
you can set a network variable based on which network is at the start of the method.
This will simplify this code block by removing unnecessary try..catch block and other duplication.